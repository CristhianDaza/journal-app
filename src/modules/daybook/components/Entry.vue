<template>
  <div
    class="entry-container pb-3 pointer p-2"
  >
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">04</span>
      <span class="mx-1 fs-5">Septiembre</span>
      <span class="mx-2 fw-light">2021, sabado</span>
    </div>
    <div class="entry-description">
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab beatae, commodi cumque ea eveniet iusto nostrum placeat porro quae, quaerat quas recusandae sapiente sunt tenetur unde. Blanditiis illo itaque odio.
    </div>
  </div>
</template>

<script>
export default {
  name: 'Entry'
}
</script>

<style lang="scss" scoped>
  .entry-container {
    border-bottom: 1px solid #2C3E50;
    transition: .2s all ease-in;

    &:hover {
      background-color: lighten($color: grey, $amount: 45);
    }

    .entry-description {
      font-size: 12px;
    }

    .entry-title {
      align-items: baseline;
    }
  }
</style>
