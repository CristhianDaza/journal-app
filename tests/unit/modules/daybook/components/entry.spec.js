import { shallowMount } from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry'
import { entry } from '../../../mock-data/test-journal-state'

describe('Pruebas en el Entry.vue', () => {
  const mockRouter = {
    push: jest.fn()
  }
  const wrapper = shallowMount(Entry, {
    props: {
      entry
    },
    global: {
      mocks: {
        $router: mockRouter
      }
    }
  })

  test('debe de hacer match con el snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe de redireccionar al hacer click en el entry-container', () => {
    const entryContainer = wrapper.find('.entry-container')
    entryContainer.trigger('click')
    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry',
      params: {
        id: entry.id
      }
    })
  })

  test('pruebas en las propiedades computadas', () => {
    expect(wrapper.vm.day).toBe(21)
    expect(wrapper.vm.mount).toBe('Septiembre')
    expect(wrapper.vm.yearDay).toBe('2021, Martes')
  })
})
