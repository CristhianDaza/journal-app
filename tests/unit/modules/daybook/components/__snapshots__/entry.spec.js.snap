// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry.vue debe de hacer match con el snapshot 1`] = `
<div class="entry-container pb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">21</span><span class="mx-1 fs-5">Septiembre</span><span class="mx-2 fw-light">2021, Martes</span></div>
  <div class="entry-description">Entrada mockeada</div>
</div>
`;
